apis/Google.LongRunning/Google.LongRunning/Operation.cs

```diff
@@ -130,16 +139,32 @@ public T Result
         /// <param name="pollSettings">The settings to use for repeated polling, or null
         /// to use the default poll settings (poll once every 10 seconds, forever).</param>
         /// <param name="callSettings">The call settings to apply on each call, or null for default settings.</param>
+        /// <param name="metadataCallback">The callback to invoke whenever a polled result contains metadata, or null
+        /// for no callback.</param>
         /// <returns>The completed operation, which can then be checked for errors or a result.</returns>
-        public Operation<T> PollUntilCompleted(PollSettings pollSettings = null, CallSettings callSettings = null)
+        public Operation<TResponse, TMetadata> PollUntilCompleted(
+            PollSettings pollSettings = null,
+            CallSettings callSettings = null,
+            Action<TMetadata> metadataCallback = null)
         {
             if (IsCompleted)
             {
                 return this;
             }
             callSettings = Client.GetEffectiveCallSettingsForGetOperation(callSettings);
 
-            Func<DateTime?, Operation<T>> pollAction = deadline => PollOnce(callSettings.WithEarlierDeadline(deadline, Client.Clock));
+            Func<DateTime?, Operation<TResponse, TMetadata>> pollAction =
+                deadline =>
```

Comment:
Defaulting to an empty method means unpacking (deserializing) the metadata even if we're just going to throw it away, which I'd rather not do. We could generate a different poll action, or convert the `Action<TMetadata>` into an `Action<Operation>` that unpacks the metadata... I think it may just be simplest to make it a nested if statement though.